### PR TITLE
[File conflict][Passing CIs] softmax_cross_entropy bfloat16

### PR DESCRIPTION
**Pull request by pawelpiotrowicz**
### PR types
 Others 

### PR changes
 Others 

### Describe
Softmax with cross-entropy bfloat16 support

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_softmax_with_cross_entropy_op.py [hunk 1]
```diff
@@ -710,5 +730,62 @@ def initParams(self):
         self.use_softmax = True
 
 
+class TestSoftmaxWithCrossEntropyOpBF16New(TestSoftmaxWithCrossEntropyOp):
+    def initParams(self):
+        self.op_type = "softmax_with_cross_entropy"
+        self.numeric_stable_mode = False
+        self.soft_label = False
+        #self.shape = [3, 5, 7, 11]
+        #self.shape = [1 , 3]
```
Comment (jakpiase) [inline python/paddle/fluid/tests/unittests/test_softmax_with_cross_entropy_op.py]: Please remove these commented lines

File: python/paddle/fluid/tests/unittests/test_softmax_with_cross_entropy_op.py [hunk 2]
```diff
@@ -710,5 +730,62 @@ def initParams(self):
...
+        self.dtype = np.uint16
+
+    def setUp(self):
+        self.initParams()
+        self.op_type = "softmax_with_cross_entropy"
+
+        # NOTE: numpy float16 have very low accuracy, use float32 for numpy check.
+        date_type = np.float32 if core.is_compiled_with_rocm() else np.float64
+        rr = np.random.uniform(0.1, 1.0, self.shape).astype(date_type)
+        print(f'Random={rr}')
```
Comment (jakpiase) [inline python/paddle/fluid/tests/unittests/test_softmax_with_cross_entropy_op.py]: Please, remove this print

**Comment by lidanqing-vv** [comment]
@pawelpiotrowicz Hi this PR has conflict. What is the status of this PR? Do you need help from Baidu side?

**Comment by pawelpiotrowicz** [comment]
Hi @sneaxiy your last PR has conflict with my code, I prepared resolve branch [pawepiot_softmax_crossentropy_bf16_resolve](https://github.com/pawelpiotrowicz/Paddle/compare/pawepiot_softmax_crossentropy_bf16_resolve) . I can't solve one issue with lables. Do you have any proposal how to merge both things ? It looks like that label has bfloat16 type so it's wrong.    
```
/backup/work/Paddle/paddle/fluid/operators/softmax_with_cross_entropy_op.h:265:23: error: no match for ‘operator!=’ (operand types are ‘int’ and ‘const phi::dtype::bfloat16’)
                 if (k != label_data[idx]) {  // label_data[idx]: this sample's label
```
